query-node/mappings/src/common.ts

```diff
@@ -0,0 +1,86 @@
+import BN from 'bn.js'
+import { SubstrateEvent } from '@dzlzv/hydra-common'
+import { DatabaseManager } from '@dzlzv/hydra-db-utils'
+import { u64 } from '@polkadot/types/primitive';
+
+import { Network } from 'query-node/src/modules/enums/enums'
+
+// Asset
+import {
+  DataObjectOwner,
+} from 'query-node/src/modules/variants/variants.model'
+import {
+  DataObject,
+  LiaisonJudgement,
+} from 'query-node/src/modules/data-object/data-object.model'
+import {
+  ContentParameters
+} from '@joystream/types/augment'
+
+const currentNetwork = Network.BABYLON
+
+/*
+  Reports that insurmountable inconsistent state has been encountered and throws an exception.
+*/
+export function inconsistentState(extraInfo: string, data?: unknown): never {
+  const errorMessage = 'Inconsistent state: ' + extraInfo
+
+  // log error
+  logger.error(errorMessage, data)
+
+  // throw exception
+  throw errorMessage
+}
+
+/*
+  Prepares data object from content parameters.
+*/
+export async function prepareDataObject(
+  contentParameters: ContentParameters,
+  blockNumber: number,
+  owner: typeof DataObjectOwner,
+): Promise<DataObject> {
+  const dataObject = new DataObject({
+    owner,
+    addedAt: blockNumber,
+    typeId: contentParameters.type_id.toNumber(),
+    // `size` is masked by `size` special name in `Struct` so there needs to be `.get('size') as unknown as u64` to retrieve proper value
+    size: (contentParameters.get('size') as unknown as u64).toBn(),
+
+    // TODO: liason & judgement information can't be retrieved now - it's neither event parameter or extrinsic parameter
+    liaisonId: new BN(0),
+    liaisonJudgement: LiaisonJudgement.PENDING,
+
```

Comment:
Liaison and liaison judgment can't be retrieved right now from runtime module `content` extrinsic (for example, `create_channel`) because it is not part of extrinsic or event parameters. The same goes for `add_content` extrinsic of `storage` module.

This will need a little runtime change, am I right?

Reply:
There was a change to liaison and judgment very recently that affects this code part. The default state truly is `Pending`, and liaison is not set before content is accepted/rejected by any storage provider.

Great point about the naming! I'm not sure why there are two names for something that seems to be one thing (or is it?).